### PR TITLE
Make SnippetChooserBlock return proper snippet values

### PR DESCRIPTION
**Pull request by jams2**
This PR updates the `snippet` field of the grapple `SnippetChooserBlock` class so snippet values in stream fields return the snippet model's fields, rather than just a single string field.

To achieve this both the grapple `SnippetChooserBlock` and `SnippetsQuery.Mixin` classes need access to the `SnippetObjectType`, which depends on the snippets registry being populated. I have added the `SnippetTypes` class to `grapple.types.snippets` for the following purposes:
- not creating the `SnippetObjectType` class when a project has no snippets (which causes an error, as a `graphene.Union` subclass requires at least one type);
- creating the `SnippetObjectType` class lazily (i.e. not until `grapple.types.snippets.SnippetsQuery.Mixin` or `grapple.types.streamfield.register_streamfield_blocks.SnippetChooserBlock` are required, and the snippets registry has been populated)

### CODE REVIEW

File: example/home/test/test_blog.py [hunk 1]
```diff
@@ -464,6 +469,60 @@ def test_blog_body_pagechooserblock(self):
...
+                text
+            }
+        }
+        """
+        query_blocks = self.get_blocks_from_body(
+            block_type, block_query=block_query, page_id=blog_page.id
+        )
+        self.assertEqual(len(query_blocks), 1)
+        block = query_blocks[0]
+        self.assertEqual(block["blockType"], block_type)
```
Comment (ababic) [inline example/home/test/test_blog.py:495]: Question: Would we expect `get_blocks_from_body()` to ever return anything but `SnippetChooserBlock` value types when that is supplied as the first argument? I can fully understand the value of the following two assertions, but this one, I'm not so sure about.

Comment (jams2) [reply example/home/test/test_blog.py:495]: Yep, I think you're correct, it's unnecessary.

**Comment by zerolab** [comment]
@jams2 would be great to fix the CI failures ahead of Friday (when I be doing a concerted grapple review/maintenance effort)

**Comment by zerolab** [comment]
Merged in 60bf85a

**Pull request closed without merging**